airbytehq/airbyte
# Airbyte CDK: move from gradle to poetry

## What 
* Migrate from Gradle to Poetry for airbyte-cdk build
* Remove all gradle-related files 
* Remove java installation from CI pipelines and replace with installing poetry 
* Move setup.py implementation to poetry
* Move high-usage scrips to poetry `poe` tasks: 
  * Testing, linting, building (generator scripts) 
  * There is a separate PR to handle the mypy implementation since it has its own CI workflow and is more complicated with the changed files 
* Remove unused scripts: 
  * `low-code-unit-test.sh`
  * `validate-yaml-schema.sh`
* Update docs to stop referencing gradle: ⚠️ I did my best here. A lot of these docs are severely out of date and deserve a separate thorough lookover. TODO: create an issue enumerating docs that should be re looked over
